systempoweraction/src/main/java/com/stream_pi/systempoweraction/SystemPowerAction.java

```diff
@@ -0,0 +1,196 @@
+package com.stream_pi.systempoweraction;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import com.stream_pi.action_api.actionproperty.property.ListValue;
+import com.stream_pi.action_api.actionproperty.property.Property;
+import com.stream_pi.action_api.actionproperty.property.Type;
+import com.stream_pi.action_api.externalplugin.NormalAction;
+import com.stream_pi.util.platform.Platform;
+import com.stream_pi.util.exception.MinorException;
+import com.stream_pi.util.version.Version;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class SystemPowerAction extends NormalAction
+{
+    public SystemPowerAction()
+    {
+        setName("Power Action");
+        setAuthor("quimodotcom");
+        setHelpLink("https://github.com/quimodotcom/essential-actions");
+        setVersion(new Version(1,0,0));
+        setCategory("System");
+        
+        states = new ArrayList<>();
+        states.addAll(Arrays.asList(
+                new ListValue("Shutdown"),
+                new ListValue("Restart"),
+                new ListValue("Sleep")
+        ));
+    }
+    
+    private final ArrayList<ListValue> states;
+    
+    @Override
+    public void initProperties() throws MinorException
+    {
+        Property status = new Property("state", Type.LIST);
+        status.setListValue(states);
+        status.setDisplayName("Power State");
+        
+        Property customLinuxSleep = new Property("linuxsl", Type.STRING);
+        customLinuxSleep.setDisplayName("Custom Linux Sleep Command");
+        customLinuxSleep.setCanBeBlank(true);
+        
+        Property customLinuxRestart = new Property("linuxre", Type.STRING);
+        customLinuxRestart.setDisplayName("Custom Linux Restart Command");
+        customLinuxRestart.setCanBeBlank(true);
+
+        Property customLinuxShutdown = new Property("linuxsh", Type.STRING);
+        customLinuxShutdown.setDisplayName("Custom Linux Shutdown Command");
+        customLinuxShutdown.setCanBeBlank(true);
+        
+        addClientProperties(status);
+        addServerProperties(customLinuxSleep, customLinuxRestart, customLinuxShutdown);
```

Comment:
Add custom properties for windows, and if possible, macos too

Reply:
done

Reply:
Macos isn't possible as there is only one way to use power commands